app/models/ability.rb

```diff
@@ -0,0 +1,43 @@
+class Ability
+  include CanCan::Ability
+
+  def initialize(user)
+    # Define abilities for the passed in user here. For example:
+    #
+    #   user ||= User.new # guest user (not logged in)
+    #   if user.admin?
+    #     can :manage, :all
+    #   else
+    #     can :read, :all
+    #   end
+    #
+    # The first argument to `can` is the action you are giving the user
+    # permission to do.
+    # If you pass :manage it will apply to every action. Other common actions
+    # here are :read, :create, :update and :destroy.
+    #
+    # The second argument is the resource the user can perform the action on.
+    # If you pass :all it will apply to every resource. Otherwise pass a Ruby
+    # class of the resource.
+    #
+    # The third argument is an optional hash of conditions to further filter the
+    # objects.
+    # For example, here the user can only update published articles.
+    #
+    #   can :update, Article, :published => true
+    #
+    # See the wiki for details:
+    # https://github.com/ryanb/cancan/wiki/Defining-Abilities
+
+    user ||= User.new # guest user (not logged in)
+
+    if user.role == 'admin'
+      can :manage, :all # admin can do everything
+    else
+      can :read, :all # everyone can read everything
+      can :create, Comment # everyone can create comments
+      can :destroy, Comment, author_id: user.id # users can delete their own comments
+      can :destroy, Post, author_id: user.id # users can delete their own posts
```

Comment:
- [OPTIONAL] You can also consider adding permission on the Post create action as you did with the Comment on line 38.